packages/bpk-component-breakpoint/README.md

```diff
@@ -24,17 +24,57 @@ export default () => (
 
 ### Server Side Render (SSR) Support
 
-You can pass `matchSSR` which will instruct the breakpoint to match any time it is rendered on the server.
+You can pass `matchSSR` which will instruct the breakpoint to match any time it is rendered on the server. You can use
+this to estimate what breakpoint is likely to match on the client-side.
 
 ```js
-<BpkBreakpoint query={BREAKPOINTS.TABLET} matchSSR>
+import { isTablet, isMobilePhone } from 'some-device-detection';
+
+<BpkBreakpoint query={BREAKPOINTS.TABLET} matchSSR={isTablet}>
   <span>Tablet viewport is active OR we are rendering on the server-side</span>
 </BpkBreakpoint>
-<BpkBreakpoint query={BREAKPOINTS.MOBILE}>
+<BpkBreakpoint query={BREAKPOINTS.MOBILE} matchSSR={isMobilePhone}>
   <span>Mobile viewport is active AND we are rendering on the client-side</span>
 </BpkBreakpoint>
 ```
 
+If you match to a different breakpoint when rendering on the server, than what is matched to in the traveller's browser,
+then React will print a warning saying there is a mismatch.
+
+### Testing
+
+When writing tests for any components that use BpkBreakpoint, you will have to mock either the `BpkBreakpoint` component
+or the underlying `useMediaQuery`. This is because the `window.matchMedia` function that we rely on does not exist in
```

Comment:
Is it worth linking to the official jest docs on this? https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom

Do you think Backpack could provide a sample function that might make this easier for consumers? e.g. allow consumers to add a line in `jest.setup`

```
import '@skyscanner/backpack-web/bpk-breakpoint/matchMedia.mock'; 
```

Reply:
It depends. If we recommend baking a mock into the jest config, we'd have to pick if we wanted the default to be to match everything or match nothing. Which might not be that useful.

The way we are recommending here is to be explicit about which viewport you are emulating and that seems safer (albeit more faff).